pygeoapi/provider/wmts_facade.py

```diff
@@ -0,0 +1,278 @@
+
+    @property
+    def service_metadata_url(self):
+        return self._service_metadata_url
+
+    def get_tiling_schemes(self):
+
+        tile_matrix_set_links_list = [
+            TileMatrixSetEnum.WORLDCRS84QUAD.value,
+            TileMatrixSetEnum.WEBMERCATORQUAD.value
+        ]
+        tile_matrix_set_links = [
+            item for item in tile_matrix_set_links_list
+            if item.tileMatrixSet == self.options['scheme']
+        ]
+
+        return tile_matrix_set_links
+
+    def get_tiles_service(self, baseurl=None, servicepath=None,
+                          dirpath=None, tile_type=None):
+        """
+        Gets service description
+
+        :param baseurl: base URL of endpoint
+        :param servicepath: base path of URL
+        :param dirpath: directory basepath (equivalent of URL)
+        :param tile_type: tile format type
+
+        :returns: `dict` of item tile service
+        """
+
+        url = urlparse(self.data)
+        baseurl = baseurl or f'{url.scheme}://{url.netloc}'
+        tile_type = self.format_type
+        basepath = url.path.split('/{z}/{x}/{y}')[0]
+        servicepath = servicepath or f'{basepath}/tiles/{{tileMatrixSetId}}/{{tileMatrix}}/{{tileRow}}/{{tileCol}}?f={tile_type}'  # noqa
+
+        if servicepath.startswith(baseurl):
+            self._service_url = servicepath
+        else:
+            self._service_url = url_join(baseurl, servicepath)
+
+        tile_matrix_set = self.service_url.split(
+            '/{tileMatrix}/{tileRow}/{tileCol}')[0]
+        self._service_metadata_url = url_join(tile_matrix_set, 'metadata')
+
+        links = {
+            'links': [
+                {
+                    'type': self.mimetype,
+                    'rel': 'item',
+                    'title': 'This collection as image tiles',
+                    'href': self.service_url
+                },
+                {
+                    'type': 'application/json',
+                    'rel': 'describedby',
+                    'title': 'Collection metadata in TileJSON format',
+                    'href': f'{self.service_metadata_url}?f=json'
+                }
+            ]
+        }
+        return links
+
+    def get_tiles(self, layer=None, tileset=None,
+                  z=None, y=None, x=None, format_=None):
+        """
+        Gets tile
+
+        :param layer: The layer name
+        :param tileset: The tileset name
+        :param z: z index
+        :param y: y index
+        :param x: x index
+        :param format_: tile format
+
+        :returns: an image tile
+        """
+        if is_url(self.data):
+            params = {
+                'service': 'WMTS',
+                'request': 'GetTile',
+                'version': '1.0.0',
+                'format': self.format_type,
+                'layer': self.options['wmts_layer'],
+                'tileMatrixSet': self.options['wmts_tile_matrix_set'],
+                'tileMatrix': z,
```

Comment:
In WMTS this value must be one of the listed tile matrix of the select tile matrix set for the layer, so this works when tileMatrixSet indicated for pygeoapi and WMTS backend are defined in the exact same way. This is mentioned in the documentation as comment to YML configuration, but I would explain this more in the text.